// @aldabil/react-scheduler/dist/context/state/stateContext.js
import enUS from "date-fns/locale/en-US";
import { createContext } from "react";
export const defaultProps = {
    height: 600,
    month: {
        weekDays: [0, 1, 2, 3, 4, 5],
        weekStartOn: 6,
        startHour: 9,
        endHour: 17,
    },
    week: {
        weekDays: [0, 1, 2, 3, 4, 5],
        weekStartOn: 6,
        startHour: 9,
        endHour: 17,
        step: 60,
    },
    day: {
        startHour: 9,
        endHour: 17,
        step: 60,
    },
    view: "week",
    selectedDate: new Date(),
    events: [],
    remoteEvents: undefined,
    fields: [],
    loading: undefined,
    customEditor: undefined,
    onConfirm: undefined,
    onDelete: undefined,
    viewerExtraComponent: undefined,
    resources: [],
    resourceFields: {
        idField: "assignee",
        textField: "text",
        subTextField: "subtext",
        avatarField: "avatar",
        colorField: "color",
    },
    recourseHeaderComponent: undefined,
    resourceViewMode: "default",
    direction: "ltr",
    dialogMaxWidth: "md",
    locale: enUS,
};
const StateContext = createContext(Object.assign(Object.assign({}, defaultProps), { mounted: false, dialog: false, selectedRange: undefined, selectedEvent: undefined, selectedResource: undefined, handleState: () => { }, getViews: () => [], triggerDialog: () => { }, triggerLoading: () => { }, handleGotoDay: () => { }, confirmEvent: () => { }, onDrop: () => { } }));
export { StateContext };
