// @mantine/core/cjs/components/NumberInput/NumberInput.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var React = require('react');
var hooks = require('@mantine/hooks');
var TextInput = require('../TextInput/TextInput.js');
var NumberInput_styles = require('./NumberInput.styles.js');

function _interopDefaultLegacy (e) { return e && typeof e === 'object' && 'default' in e ? e['default'] : e; }

var React__default = /*#__PURE__*/_interopDefaultLegacy(React);

var __defProp = Object.defineProperty;
var __defProps = Object.defineProperties;
var __getOwnPropDescs = Object.getOwnPropertyDescriptors;
var __getOwnPropSymbols = Object.getOwnPropertySymbols;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __propIsEnum = Object.prototype.propertyIsEnumerable;
var __defNormalProp = (obj, key, value) => key in obj ? __defProp(obj, key, { enumerable: true, configurable: true, writable: true, value }) : obj[key] = value;
var __spreadValues = (a, b) => {
  for (var prop in b || (b = {}))
    if (__hasOwnProp.call(b, prop))
      __defNormalProp(a, prop, b[prop]);
  if (__getOwnPropSymbols)
    for (var prop of __getOwnPropSymbols(b)) {
      if (__propIsEnum.call(b, prop))
        __defNormalProp(a, prop, b[prop]);
    }
  return a;
};
var __spreadProps = (a, b) => __defProps(a, __getOwnPropDescs(b));
var __objRest = (source, exclude) => {
  var target = {};
  for (var prop in source)
    if (__hasOwnProp.call(source, prop) && exclude.indexOf(prop) < 0)
      target[prop] = source[prop];
  if (source != null && __getOwnPropSymbols)
    for (var prop of __getOwnPropSymbols(source)) {
      if (exclude.indexOf(prop) < 0 && __propIsEnum.call(source, prop))
        target[prop] = source[prop];
    }
  return target;
};
const NumberInput = React.forwardRef((_a, ref) => {
  var _b = _a, {
    disabled,
    value,
    onChange,
    min,
    max,
    step = 1,
    onBlur,
    onFocus,
    hideControls = false,
    radius = "sm",
    variant,
    precision = 0,
    defaultValue,
    noClampOnBlur = false,
    handlersRef,
    classNames,
    styles,
    size
  } = _b, others = __objRest(_b, [
    "disabled",
    "value",
    "onChange",
    "min",
    "max",
    "step",
    "onBlur",
    "onFocus",
    "hideControls",
    "radius",
    "variant",
    "precision",
    "defaultValue",
    "noClampOnBlur",
    "handlersRef",
    "classNames",
    "styles",
    "size"
  ]);
  const { classes, cx, theme } = NumberInput_styles['default']({ radius, size }, { classNames, styles, name: "NumberInput" });
  const [focused, setFocused] = React.useState(false);
  const [_value, setValue] = React.useState(typeof value === "number" ? value : typeof defaultValue === "number" ? defaultValue : void 0);
  const finalValue = typeof value === "number" ? value : _value;
  const [tempValue, setTempValue] = React.useState(typeof finalValue === "number" ? finalValue.toFixed(precision) : "");
  const inputRef = React.useRef();
  const handleValueChange = (val) => {
    typeof onChange === "function" && onChange(val);
    setValue(val);
  };
  const _min = typeof min === "number" ? min : -Infinity;
  const _max = typeof max === "number" ? max : Infinity;
  const increment = () => {
    var _a2;
    if (_value === void 0) {
      handleValueChange(min != null ? min : 0);
      setTempValue((_a2 = min == null ? void 0 : min.toFixed(precision)) != null ? _a2 : "0");
    } else {
      const result = hooks.clamp({ value: _value + step, min: _min, max: _max }).toFixed(precision);
      handleValueChange(parseFloat(result));
      setTempValue(result);
    }
  };
  const decrement = () => {
    var _a2;
    if (_value === void 0) {
      handleValueChange(min != null ? min : 0);
      setTempValue((_a2 = min == null ? void 0 : min.toFixed(precision)) != null ? _a2 : "0");
    } else {
      const result = hooks.clamp({ value: _value - step, min: _min, max: _max }).toFixed(precision);
      handleValueChange(parseFloat(result));
      setTempValue(result);
    }
  };
  hooks.assignRef(handlersRef, { increment, decrement });
  React.useEffect(() => {
    if (typeof value === "number" && !focused) {
      setValue(value);
      setTempValue(value.toFixed(precision));
    }
    if (defaultValue === void 0 && value === void 0 && !focused) {
      setValue(value);
      setTempValue("");
    }
  }, [value]);
  const rightSection = /* @__PURE__ */ React__default.createElement("div", {
    className: classes.rightSection
  }, /* @__PURE__ */ React__default.createElement("button", {
    type: "button",
    tabIndex: -1,
    "aria-hidden": true,
    disabled: finalValue >= max,
    className: cx(classes.control, classes.controlUp),
    onMouseDown: (event) => {
      event.preventDefault();
      increment();
      inputRef.current.focus();
    }
  }), /* @__PURE__ */ React__default.createElement("button", {
    type: "button",
    tabIndex: -1,
    "aria-hidden": true,
    disabled: finalValue <= min,
    className: cx(classes.control, classes.controlDown),
    onMouseDown: (event) => {
      event.preventDefault();
      decrement();
      inputRef.current.focus();
    }
  }));
  const handleChange = (event) => {
    const val = event.target.value;
    setTempValue(val);
    if (val === "") {
      handleValueChange(void 0);
    } else {
      const parsed = Number(val);
      val.trim() !== "" && !Number.isNaN(parsed) && handleValueChange(parsed);
    }
  };
  const handleBlur = (event) => {
    var _a2;
    if (event.target.value === "") {
      setTempValue("");
      handleValueChange(void 0);
    } else {
      const parsedVal = parseFloat(event.target.value);
      const val = hooks.clamp({ value: parsedVal, min: _min, max: _max });
      if (!Number.isNaN(val)) {
        if (!noClampOnBlur) {
          setTempValue(val.toFixed(precision));
          handleValueChange(val);
        }
      } else {
        setTempValue((_a2 = finalValue == null ? void 0 : finalValue.toFixed(precision)) != null ? _a2 : "");
      }
    }
    setFocused(false);
    typeof onBlur === "function" && onBlur(event);
  };
  const handleFocus = (event) => {
    setFocused(true);
    typeof onFocus === "function" && onFocus(event);
  };
  const handleKeyDown = (event) => {
    if (event.key === "ArrowUp") {
      event.preventDefault();
      increment();
    } else if (event.key === "ArrowDown") {
      decrement();
    }
  };
  return /* @__PURE__ */ React__default.createElement(TextInput.TextInput, __spreadProps(__spreadValues({}, others), {
    variant,
    value: tempValue,
    disabled,
    ref: hooks.useMergedRef(inputRef, ref),
    type: "text",
    onChange: handleChange,
    onBlur: handleBlur,
    onFocus: handleFocus,
    onKeyDown: handleKeyDown,
    rightSection: disabled || hideControls || variant === "unstyled" ? null : rightSection,
    rightSectionWidth: theme.fn.size({ size, sizes: NumberInput_styles.CONTROL_SIZES }) + 1,
    radius,
    max,
    min,
    step,
    size,
    styles,
    classNames,
    __staticSelector: "NumberInput"
  }));
});
NumberInput.displayName = "@mantine/core/NumberInput";

exports.NumberInput = NumberInput;
