// @mantine/core/cjs/components/Select/SelectDropdown/SelectDropdown.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var React = require('react');
var SelectDropdown_styles = require('./SelectDropdown.styles.js');
var Popper = require('../../Popper/Popper.js');
var Paper = require('../../Paper/Paper.js');

function _interopDefaultLegacy (e) { return e && typeof e === 'object' && 'default' in e ? e['default'] : e; }

var React__default = /*#__PURE__*/_interopDefaultLegacy(React);

const SelectDropdown = React.forwardRef(({
  mounted,
  transition,
  transitionDuration,
  transitionTimingFunction,
  uuid,
  shadow,
  maxDropdownHeight,
  children,
  classNames,
  styles,
  dropdownComponent,
  referenceElement,
  direction = "column",
  onDirectionChange,
  zIndex = 2e3,
  __staticSelector
}, ref) => {
  const { classes } = SelectDropdown_styles['default'](null, { classNames, styles, name: __staticSelector });
  return /* @__PURE__ */ React__default.createElement(Popper.Popper, {
    referenceElement,
    mounted,
    transition,
    transitionDuration,
    transitionTimingFunction,
    position: "bottom",
    placementFallbacks: ["top"],
    zIndex,
    onPlacementChange: (placement) => {
      const nextDirection = placement === "top" ? "column-reverse" : "column";
      if (direction !== nextDirection) {
        onDirectionChange && onDirectionChange(nextDirection);
      }
    },
    modifiers: [
      {
        name: "sameWidth",
        enabled: true,
        phase: "beforeWrite",
        requires: ["computeStyles"],
        fn: ({ state }) => {
          state.styles.popper.width = `${state.rects.reference.width}px`;
        },
        effect: ({ state }) => {
          state.elements.popper.style.width = `${state.elements.reference.offsetWidth}px`;
        }
      }
    ]
  }, /* @__PURE__ */ React__default.createElement(Paper.Paper, {
    component: dropdownComponent || "div",
    id: `${uuid}-items`,
    "aria-labelledby": `${uuid}-label`,
    role: "listbox",
    className: classes.dropdown,
    shadow,
    ref,
    style: {
      maxHeight: maxDropdownHeight,
      flexDirection: direction
    },
    onMouseDown: (event) => event.preventDefault()
  }, children));
});
SelectDropdown.displayName = "@mantine/core/SelectDropdown";

exports.SelectDropdown = SelectDropdown;
