// @mantine/core/cjs/components/Stepper/Stepper.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var React = require('react');
var styles = require('@mantine/styles');
var Step = require('./Step/Step.js');
var Stepper_styles = require('./Stepper.styles.js');

function _interopDefaultLegacy (e) { return e && typeof e === 'object' && 'default' in e ? e['default'] : e; }

var React__default = /*#__PURE__*/_interopDefaultLegacy(React);

var __defProp = Object.defineProperty;
var __defProps = Object.defineProperties;
var __getOwnPropDescs = Object.getOwnPropertyDescriptors;
var __getOwnPropSymbols = Object.getOwnPropertySymbols;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __propIsEnum = Object.prototype.propertyIsEnumerable;
var __defNormalProp = (obj, key, value) => key in obj ? __defProp(obj, key, { enumerable: true, configurable: true, writable: true, value }) : obj[key] = value;
var __spreadValues = (a, b) => {
  for (var prop in b || (b = {}))
    if (__hasOwnProp.call(b, prop))
      __defNormalProp(a, prop, b[prop]);
  if (__getOwnPropSymbols)
    for (var prop of __getOwnPropSymbols(b)) {
      if (__propIsEnum.call(b, prop))
        __defNormalProp(a, prop, b[prop]);
    }
  return a;
};
var __spreadProps = (a, b) => __defProps(a, __getOwnPropDescs(b));
var __objRest = (source, exclude) => {
  var target = {};
  for (var prop in source)
    if (__hasOwnProp.call(source, prop) && exclude.indexOf(prop) < 0)
      target[prop] = source[prop];
  if (source != null && __getOwnPropSymbols)
    for (var prop of __getOwnPropSymbols(source)) {
      if (exclude.indexOf(prop) < 0 && __propIsEnum.call(source, prop))
        target[prop] = source[prop];
    }
  return target;
};
const Stepper = React.forwardRef((_a, ref) => {
  var _b = _a, {
    className,
    children,
    onStepClick,
    active,
    completedIcon,
    progressIcon,
    color,
    iconSize,
    style,
    contentPadding = "md",
    size = "md",
    orientation = "horizontal",
    breakpoint,
    iconPosition = "left",
    classNames,
    styles: styles$1,
    sx
  } = _b, others = __objRest(_b, [
    "className",
    "children",
    "onStepClick",
    "active",
    "completedIcon",
    "progressIcon",
    "color",
    "iconSize",
    "style",
    "contentPadding",
    "size",
    "orientation",
    "breakpoint",
    "iconPosition",
    "classNames",
    "styles",
    "sx"
  ]);
  var _a2, _b2;
  const { mergedStyles, rest } = styles.useExtractedMargins({ others, style });
  const { classes, cx } = Stepper_styles['default']({ contentPadding, color, orientation, iconPosition, size, iconSize, breakpoint }, { classNames, styles: styles$1, sx, name: "Stepper" });
  const filteredChildren = React.Children.toArray(children).filter((item) => item.type === Step.Step);
  const items = filteredChildren.reduce((acc, item, index, array) => {
    acc.push(/* @__PURE__ */ React__default.createElement(Step.Step, __spreadProps(__spreadValues({}, item.props), {
      __staticSelector: "Stepper",
      icon: item.props.icon || index + 1,
      key: index,
      state: active === index ? "stepProgress" : active > index ? "stepCompleted" : "stepInactive",
      onClick: () => typeof onStepClick === "function" && onStepClick(index),
      allowStepClick: typeof onStepClick === "function",
      completedIcon: item.props.completedIcon || completedIcon,
      progressIcon: item.props.progressIcon || progressIcon,
      color: item.props.color || color,
      iconSize,
      size,
      classNames,
      styles: styles$1,
      iconPosition: item.props.iconPosition || iconPosition
    })));
    if (index !== array.length - 1) {
      acc.push(/* @__PURE__ */ React__default.createElement("div", {
        className: cx(classes.separator, { [classes.separatorActive]: index < active }),
        key: `separator-${index}`
      }));
    }
    return acc;
  }, []);
  const content = (_b2 = (_a2 = filteredChildren[active]) == null ? void 0 : _a2.props) == null ? void 0 : _b2.children;
  return /* @__PURE__ */ React__default.createElement("div", __spreadValues({
    className: cx(classes.root, className),
    style: mergedStyles,
    ref
  }, rest), /* @__PURE__ */ React__default.createElement("div", {
    className: classes.steps
  }, items), content && /* @__PURE__ */ React__default.createElement("div", {
    className: classes.content
  }, content));
});
Stepper.Step = Step.Step;
Stepper.displayName = "@mantine/core/Stepper";

exports.Stepper = Stepper;
